### PR TITLE
Oracle ana

### PR DESCRIPTION
**Pull request by anarojoagusti**
This NFT frontend consist of 3 main panels for interacting with the NFT minting process. 

- **Left panel:** displays the contract details, Chainlink _V_RF Coordinator,_ KeyHash, VRF ubscritiopn ID, Gas Limit_, and also the _Contract Address_ and its _owner's address._ This info could be removed from the frontend, but for now it's display for checking purposes.

- **Center panel:** provides buttons to request random words (only the contract's owner can request it, a.k.a. my address) but once the VRF request is approved and the '_requestID_' is received, the user can enter it into the textfield and click on the 'Mint NFT' button. Then, he can check his balance with the 'Check NFT balance' button (it chacks the balanceOf the address of the connected wallet). 

**Note:** when a '_requestId_' is received and used for mint a NFT, its random number is used to tag the '_serie_' of that NFT. So each token has a '_serie_', a '_name_' and an '_image_' values. By default, all the NFts have the same name and image. Ideally we should have a collection of different images/names and assign them randomly, but right now it's not that fancy.

- **Right panel**: allows the user to view details of a specific owned NFT, entering the token ID, and shows the series (vrf random num), the name ('idea') and the ipfs-hosted image of the NFT.

image

### CODE REVIEW

File: hardhat/contracts/NFTmintingVRF.sol [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    using VRFV2PlusClient for VRFV2PlusClient.RandomWordsRequest;
+    using Strings for uint256;
+
+    // VRF Coordinator
+    uint256 public subscriptionId;
+    bytes32 public keyHash;
+    uint256 public requestId;
+    uint32 public callbackGasLimit = 200000;
+    uint16 public requestConfirmations = 3;
+    uint32 public numWords = 1;
```
Comment (Cascou) [inline hardhat/contracts/NFTmintingVRF.sol:21]: suggestion: can be made as constant for gas efficiency

File: hardhat/contracts/NFTmintingVRF.sol [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    //track mint tokens
+    uint256 public tokenCounter;
+    mapping(uint256 => LabsNFT) public labsNFT;
+    mapping(uint256 => RequestStatus) public s_requests; //requestId --> requestStatus
+
+    //base image of LabsNFTs
+    string private _imageURI = "https://ipfs.io/ipfs/QmVjJtouNYv89rqPiXoE6afBsasLVjNNj87K5cJy9GzS4Y";
+
```
Comment (Cascou) [inline hardhat/contracts/NFTmintingVRF.sol:33]: suggestion: can be made as immutable (as it is a string) for gas efficiency

**Review by Dhruv-Varshney-developer — APPROVED** [review]
Looks awesome.
Please feel free to merge!

**Review by Cascou — APPROVED** [review]

**Comment by anarojoagusti** [comment]
Alert added Screenshot from 2024-10-23 20-21-41

**Pull request merged**